cloud/openstack/os_keystone_endpoint.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/python
+
+# Copyright (c) 2016 Helicom
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+from distutils.version import StrictVersion
+
+
+DOCUMENTATION = '''
+---
+module: os_api_endpoint
+short_description: Create, update or delete API endpoints.
+extends_documentation_fragment: openstack
+author: "Michel Labarre (@mlabarre)"
+version_added: "2.2"
+description:
+    - Add, update or delete API endpoints.
+    - V2.0 and V3 API version are supported. So arguments as interface,
+      url, public_url, admin_url and internal_url are supported. But
+        - for V2 version, only public_url, internal_url and admin_url
+          can be specified (not interface, neither url).
+        - for V3 version, only interface AND url can be specified
+          (no public_url, internal_url and admin_url).
```

Comment:
Is it not possible to just take in URL and interface as module arguments, and then internally or through shade decide how to represent the data to the API? The input to the module shouldn't change. The admin still has to provide a URL and some indication of what type of URL it is. I don't think the arguments to the ansible module should change depending on API version.